src/Buffer.mo

```diff
@@ -1790,6 +2496,20 @@ module {
 
   /// Groups equal and adjacent elements in the list into sub lists.
   ///
+  /// Example:
+  /// ```motoko include=initialize
+  ///
+  /// let buffer = Buffer.Buffer<Nat>(2);
+  /// buffer.add(1);
+  /// buffer.add(2);
+  /// buffer.add(3);
+  /// buffer.add(4);
+  /// buffer.add(5);
+  /// buffer.add(6);
+  ///
+  /// Buffer.groupBy<Nat>(buffer, func (x, y) { x == y });
```

Comment:
I would pull a similar trick here to the `chunks` suggestion, and anywhere else you have to display a `Buffer<Buffer<Nat>>` to the user